In [35]:
import pandas as pd
import numpy as np  
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stats
import datetime as dt

In [36]:
# Importing the data / Leitura dos dados

url_hypotheses_us_log = 'https://raw.githubusercontent.com/RafaelGenari/Bootcamp_Project_Making_Decisions_Based_In_Data/refs/heads/main/hypotheses_us.csv'

hypotheses_log = pd.read_csv(url_hypotheses_us_log, sep = ';')

url_orders_us_log ='https://raw.githubusercontent.com/RafaelGenari/Bootcamp_Project_Making_Decisions_Based_In_Data/refs/heads/main/orders_us.csv'

orders_log = pd.read_csv(url_orders_us_log, sep = ',')

url_visits_us_log = 'https://raw.githubusercontent.com/RafaelGenari/Bootcamp_Project_Making_Decisions_Based_In_Data/refs/heads/main/visits_us.csv'

visits_log = pd.read_csv(url_visits_us_log, sep = ',')

## Data Cleaning / Limpeza dos Dados

### Hyphotheses Log

In [37]:
hypotheses_log.info()
print()
hypotheses_log.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Hypothesis  9 non-null      object
 1   Reach       9 non-null      int64 
 2   Impact      9 non-null      int64 
 3   Confidence  9 non-null      int64 
 4   Effort      9 non-null      int64 
dtypes: int64(4), object(1)
memory usage: 492.0+ bytes



Hypothesis  Reach  Impact  \
0  Add two new channels for attracting traffic. T...      3      10   
1  Launch your own delivery service. This will sh...      2       5   
2  Add product recommendation blocks to the store...      8       3   
3  Change the category structure. This will incre...      8       3   
4  Change the background color on the main page. ...      3       1   

   Confidence  Effort  
0           8       6  
1           4      10  
2           7       3  
3           3       8  
4           1       1

In [38]:
hypotheses_log.describe()

Reach     Impact  Confidence     Effort
count   9.000000   9.000000    9.000000   9.000000
mean    4.777778   4.777778    5.555556   4.888889
std     3.153481   3.192874    3.045944   2.803767
min     1.000000   1.000000    1.000000   1.000000
25%     3.000000   3.000000    3.000000   3.000000
50%     3.000000   3.000000    7.000000   5.000000
75%     8.000000   7.000000    8.000000   6.000000
max    10.000000  10.000000    9.000000  10.000000

In [66]:
hypotheses_log.columns = hypotheses_log.columns.str.lower()

In [67]:
hypotheses_log.info()
print()
hypotheses_log.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   hypothesis  9 non-null      object
 1   reach       9 non-null      int64 
 2   impact      9 non-null      int64 
 3   confidence  9 non-null      int64 
 4   effort      9 non-null      int64 
dtypes: int64(4), object(1)
memory usage: 492.0+ bytes



hypothesis  reach  impact  \
0  Add two new channels for attracting traffic. T...      3      10   
1  Launch your own delivery service. This will sh...      2       5   
2  Add product recommendation blocks to the store...      8       3   
3  Change the category structure. This will incre...      8       3   
4  Change the background color on the main page. ...      3       1   

   confidence  effort  
0           8       6  
1           4      10  
2           7       3  
3           3       8  
4           1       1

In [61]:
hypotheses_log.duplicated().sum()

0

### Orders Log

In [46]:
orders_log.info()
print()
orders_log.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1197 entries, 0 to 1196
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   transaction_id  1197 non-null   int64         
 1   visitor_id      1197 non-null   int64         
 2   date            1197 non-null   datetime64[ns]
 3   revenue         1197 non-null   float64       
 4   group           1197 non-null   object        
dtypes: datetime64[ns](1), float64(1), int64(2), object(1)
memory usage: 46.9+ KB



transaction_id  visitor_id       date  revenue group
0      3667963787  3312258926 2019-08-15     30.4     B
1      2804400009  3642806036 2019-08-15     15.2     B
2      2961555356  4069496402 2019-08-15     10.2     A
3      3797467345  1196621759 2019-08-15    155.1     B
4      2282983706  2322279887 2019-08-15     40.5     B

In [48]:
orders_log.describe()

transaction_id    visitor_id                           date  \
count    1.197000e+03  1.197000e+03                           1197   
mean     2.155621e+09  2.165960e+09  2019-08-15 13:06:46.015037696   
min      1.062393e+06  5.114589e+06            2019-08-01 00:00:00   
25%      1.166776e+09  1.111826e+09            2019-08-08 00:00:00   
50%      2.145194e+09  2.217985e+09            2019-08-15 00:00:00   
75%      3.237740e+09  3.177606e+09            2019-08-23 00:00:00   
max      4.293856e+09  4.283872e+09            2019-08-31 00:00:00   
std      1.229085e+09  1.236014e+09                            NaN   

            revenue  
count   1197.000000  
mean     131.491646  
min        5.000000  
25%       20.800000  
50%       50.200000  
75%      130.400000  
max    19920.400000  
std      603.004729

In [49]:
orders_log['date'] = pd.to_datetime(orders_log['date'])

In [50]:
orders_log = orders_log.rename(columns = {'transactionId': 'transaction_id', 'visitorId': 'visitor_id'})

In [51]:
orders_log.info()
print()
orders_log.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1197 entries, 0 to 1196
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   transaction_id  1197 non-null   int64         
 1   visitor_id      1197 non-null   int64         
 2   date            1197 non-null   datetime64[ns]
 3   revenue         1197 non-null   float64       
 4   group           1197 non-null   object        
dtypes: datetime64[ns](1), float64(1), int64(2), object(1)
memory usage: 46.9+ KB



transaction_id  visitor_id       date  revenue group
0      3667963787  3312258926 2019-08-15     30.4     B
1      2804400009  3642806036 2019-08-15     15.2     B
2      2961555356  4069496402 2019-08-15     10.2     A
3      3797467345  1196621759 2019-08-15    155.1     B
4      2282983706  2322279887 2019-08-15     40.5     B

In [63]:
orders_log.duplicated().sum()

0

### Visits Log

In [54]:
visits_log.info()
print()
visits_log.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62 entries, 0 to 61
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   date    62 non-null     datetime64[ns]
 1   group   62 non-null     object        
 2   visits  62 non-null     int64         
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 1.6+ KB



date group  visits
0 2019-08-01     A     719
1 2019-08-02     A     619
2 2019-08-03     A     507
3 2019-08-04     A     717
4 2019-08-05     A     756

In [55]:
orders_log.describe()

transaction_id    visitor_id                           date  \
count    1.197000e+03  1.197000e+03                           1197   
mean     2.155621e+09  2.165960e+09  2019-08-15 13:06:46.015037696   
min      1.062393e+06  5.114589e+06            2019-08-01 00:00:00   
25%      1.166776e+09  1.111826e+09            2019-08-08 00:00:00   
50%      2.145194e+09  2.217985e+09            2019-08-15 00:00:00   
75%      3.237740e+09  3.177606e+09            2019-08-23 00:00:00   
max      4.293856e+09  4.283872e+09            2019-08-31 00:00:00   
std      1.229085e+09  1.236014e+09                            NaN   

            revenue  
count   1197.000000  
mean     131.491646  
min        5.000000  
25%       20.800000  
50%       50.200000  
75%      130.400000  
max    19920.400000  
std      603.004729

In [56]:
visits_log['date'] = pd.to_datetime(visits_log['date'])

In [57]:
visits_log.info()
print()
visits_log.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62 entries, 0 to 61
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   date    62 non-null     datetime64[ns]
 1   group   62 non-null     object        
 2   visits  62 non-null     int64         
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 1.6+ KB



date group  visits
0 2019-08-01     A     719
1 2019-08-02     A     619
2 2019-08-03     A     507
3 2019-08-04     A     717
4 2019-08-05     A     756

In [64]:
visits_log.duplicated().sum()

0

## Data Analysis / Análise dos Dados

In [68]:
hypotheses_log['ICE'] =(
                        hypotheses_log['impact']
                        * hypotheses_log['confidence']
) / hypotheses_log['effort']

hypotheses_log.head()

hypothesis  reach  impact  \
0  Add two new channels for attracting traffic. T...      3      10   
1  Launch your own delivery service. This will sh...      2       5   
2  Add product recommendation blocks to the store...      8       3   
3  Change the category structure. This will incre...      8       3   
4  Change the background color on the main page. ...      3       1   

   confidence  effort        ICE  
0           8       6  13.333333  
1           4      10   2.000000  
2           7       3   7.000000  
3           3       8   1.125000  
4           1       1   1.000000

In [70]:
hypotheses_log['RICE'] =(
                        hypotheses_log['reach']
                        * hypotheses_log['impact']
                        * hypotheses_log['confidence']
) / hypotheses_log['effort']

hypotheses_log

hypothesis  reach  impact  \
0  Add two new channels for attracting traffic. T...      3      10   
1  Launch your own delivery service. This will sh...      2       5   
2  Add product recommendation blocks to the store...      8       3   
3  Change the category structure. This will incre...      8       3   
4  Change the background color on the main page. ...      3       1   
5  Add a customer review page. This will increase...      3       2   
6  Show banners with current offers and sales on ...      5       3   
7  Add a subscription form to all the main pages....     10       7   
8  Launch a promotion that gives users discounts ...      1       9   

   confidence  effort        ICE   RICE  
0           8       6  13.333333   40.0  
1           4      10   2.000000    4.0  
2           7       3   7.000000   56.0  
3           3       8   1.125000    9.0  
4           1       1   1.000000    3.0  
5           2       3   1.333333    4.0  
6           8       3   8.000000   40.0  
7           8       5  11.200000  112.0  
8           9       5  16.200000   16.2

### Conclusion / Conslusão

Podemos reparar que hipóteses com alto alcance, coluna 'reach' com número próximo a 10 (valor máximo), tendem a ter uma prioridade maior em RICE do que em ICE, como por exemplo a hipótese 7 que tem um valor de 10 em 'reach' e consequentemente o valor de RICE (112.0) é muito 10x maior que o valor de ICE (11.20).

Quando o valor da coluna de alcance ('reach') tem um valor baixo, os valores da coluna RICE e ICE a ser semelhante ou até mesmo igual, como por exemplo a hipótese 8 que tem o valor de 1 em 'reach' e os valores de RICE E ICE é igual (16.2) ou até mesmo a hipótese 1 que tem um 'reach' de 2 e uma diferença mínima entre RICE (4.0) e ICE (2.0)

Essas variações podem acontecer porque ao incluir o fator 'reach' no cálculo do método, irá beneficia alterações que afetam um número maior de usuários, independentemente do esforço ou do impacto individual. Em contrapartida, o ICE prioriza apenas a combinação de impacto, confiança e esforço, ignorando o número de pessoas impactadas.